src/extension.ts

```diff
@@ -142,6 +149,22 @@ export async function activate(context: vscode.ExtensionContext): Promise<vscode
 	return Promise.resolve(context);
 }
 
+export async function _isApexGuruEnabledInOrg(): Promise<boolean> {
+	try {
+		const connection = await CoreExtensionService.getConnection();
+		const response:ApexGuruAuthResponse = await connection.request({
+			method: 'GET',
+			url: Constants.APEX_GURU_AUTH_ENDPOINT,
+			body: ''
+		});
+		return response.status == 'Success';
+	} catch(e) {
+		// This could throw an error for a variety of reasons. The API endpoint has not been deployed to the instance, org has no perms, timeouts etc,.
+		// In all of these scenarios, we return false.
+		return false;
```

Comment:
This would at the very least be a nice thing to log at some point ... so that if we can't figure out what went wrong - we have somewhere to look.

Reply:
Yes, I really wanted to add a log here as well. But since this was the first step in the activation (before the output channels are defined), I decided not to. May be I should move this to a later point and enable some logging.

Reply:
Took care of this, along with some refactoring to make logging a field instead of being passed around.